src/mme-app/actionHandlers/dedBearerDeactProcedureActionHandlers.cpp

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright 2020-present Infosys Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/******************************************************************************
+ *
+ * This file has both generated and manual code.
+ * 
+ * File template used for code generation:
+ * <TOP-DIR/scripts/SMCodeGen/templates/stateMachineTmpls/actionHandlers.cpp.tt>
+ *
+ ******************************************************************************/
+#include "actionHandlers/actionHandlers.h"
+#include "contextManager/subsDataGroupManager.h"
+#include "contextManager/dataBlocks.h"
+#include "controlBlock.h" 
+#include "err_codes.h"
+#include "gtpCauseTypes.h"
+#include <interfaces/mmeIpcInterface.h>
+#include <ipcTypes.h>
+#include "log.h"
+#include "mme_app.h"
+#include <mmeSmDefs.h>
+#include <msgBuffer.h>
+#include "msgType.h"
+#include "procedureStats.h"
+#include "state.h"
+#include <stateMachineEngine.h>
+#include <tipcTypes.h>
+#include <utils/mmeCauseUtils.h>
+#include <utils/mmeContextManagerUtils.h>
+#include <utils/mmeS1MsgUtils.h>
+#include "mmeStatsPromClient.h"
+
+using namespace mme;
+using namespace SM;
+using namespace cmn;
+using namespace cmn::utils;
+
+/***************************************
+ * Action handler : process_erab_release_response
+ ***************************************/
+ActStatus ActionHandlers::process_erab_release_response(ControlBlock &cb)
+{
+    log_msg(LOG_DEBUG, "process_erab_release_response : Entry\n");
+
+    UEContext *ueCtxt_p = static_cast<UEContext*>(cb.getPermDataBlock());
+    VERIFY_UE(cb, ueCtxt_p, "Invalid UE\n");
+
+    SmDedDeActProcCtxt *dedBrDeActProc_p =
+            dynamic_cast<SmDedDeActProcCtxt*>(cb.getTempDataBlock());
+    VERIFY(dedBrDeActProc_p, return ActStatus::ABORT,
+            "Ded Bearer DeActivation Procedure Context is NULL \n");
+
+    MsgBuffer *msgBuf = static_cast<MsgBuffer*>(cb.getMsgData());
+    VERIFY(msgBuf,
+            dedBrDeActProc_p->setMmeErrorCause(INVALID_MSG_BUFFER); return ActStatus::ABORT,
+            "process_erab_release_response : Invalid message buffer \n");
+
+    const s1_incoming_msg_data_t *s1_msg_data =
+            static_cast<const s1_incoming_msg_data_t*>(msgBuf->getDataPointer());
+    VERIFY(s1_msg_data,
+            dedBrDeActProc_p->setMmeErrorCause(INVALID_DATA_BUFFER); return ActStatus::ABORT,
+            "process_erab_release_response : Invalid data buffer \n");
+
+    const struct erab_rel_resp_Q_msg &erab_rel_resp =
+            (s1_msg_data->msg_data.erab_rel_resp_Q_msg_m);
+
+    ActStatus actStatus = ActStatus::PROCEED;
+    bool bearerEntryFound = false;
+
+    for (int i = 0; i < erab_rel_resp.erab_rel_list.count; i++)
+    {
+        if (erab_rel_resp.erab_rel_list.erab_id[i]
+                == dedBrDeActProc_p->getBearerId())
+        {
```

Comment:
add comment if something needs to be done in future. What if one bearer found but other bearers are not found ?